app/models/friend_list.rb

```diff
@@ -1,5 +1,59 @@
 class FriendList < ApplicationRecord
 
   belongs_to :user
-  belongs_to :friend, class_name: "User", foreign_key: 'friend_id', dependent: :destroy
+  belongs_to :friend, class_name: "User", foreign_key: 'friend_id'
+
+  def self.current_user_friend_lists(user)
+    user.friends
+        .select("friend_lists.id AS id, friend_lists.user_id, users.id AS friend_id, users.full_name, users.level")
+        .order("users.full_name")
+  end
+
+  def self.add_friend_in_friend_lists(current_user, friend_id)
+    friend = User.find_by(id: friend_id)
+
+    if friend.nil?
+      return { status: :not_found, message: "The user you are trying to add does not exist" }
+    end
```

Comment:
Is there any possibility of being a friend empty? Ultimately you are sending a friend request to a user. If there are no users you can not send friend reqeust.

Reply:
The reason I added this check is to account for a situation where a user sends a friend request, and the receiver sees it in their request panel but, before accepting, the sender deletes their account.